src/Events/Services/AppEventsService.cs

```diff
@@ -66,9 +66,17 @@ public async Task<string> StoreCloudEvent(CloudEvent cloudEvent)
         /// <inheritdoc/>
         public async Task<List<CloudEvent>> GetAppEvents(string after, DateTime? from, DateTime? to, int partyId, List<string> source, List<string> type, string unit, string person, int size = 50)
         {
-            if (partyId <= 0)
+            if ((!string.IsNullOrEmpty(person) || !string.IsNullOrEmpty(unit)) && partyId <= 0)
             {
-                partyId = await _registerService.PartyLookup(unit, person);
+                try
+                {
```

Comment:
still not sure about this try/catch.. Given that registerService throws the same exception, wouldn't this bubble up to the controller in the first place and not continue the flow further?

Reply:
With the current code I would consider the try-catch to be unnecessary. It does nothing and has no effect.